Tests/API/Controllers/TraderManagementControllerTests.cs

```diff
@@ -0,0 +1,71 @@
+
+using Microsoft.AspNetCore.Mvc;
+using Application.Interfaces;
+using NUnit.Framework;
+using API.Controllers;
+using Domain.Entities;
+using Moq;
+
+
+namespace Tests.API.Controllers
+{
+    [TestFixture]
+    public class TraderManagementControllerTests
+    {
+        private Mock<ITraderManagement> _mockTraderManagement = null!;
+        private TraderManagementController _controller = null!;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockTraderManagement = new Mock<ITraderManagement>();
+            _controller = new TraderManagementController(_mockTraderManagement.Object);
+        }
+
+        [Test]
+        public async Task AddTrader_ShouldReturnOk_WhenTraderIsAddedSuccessfully()
+        {
+            // Arrange
+            var trader = new Trader 
+            { 
+                Id = Guid.NewGuid(), 
+                Name = "John Doe", 
+                AccountBalance = 1000m 
+            };
+
+            _mockTraderManagement.Setup(x => x.AddTraderAsync(It.IsAny<Trader>()))
+                .Returns(Task.CompletedTask);
+
+            // Act
+            var result = await _controller.AddTrader(trader);
+
+            // Assert
+            Assert.That(result, Is.TypeOf<OkObjectResult>());
+            var okResult = result as OkObjectResult;
+            Assert.That(okResult?.Value, Is.EqualTo("Trader added successfully."));
+        }
+
+        [Test]
+        public async Task AddTrader_ShouldReturn400_WhenValidationFails()
```

Comment:
split the test to 2, one test for name and one test for balance. try to create method to prevent code duplication.